### PR TITLE
Hide homepage content section if empty title and content

### LINKED ISSUE #673: Hide content section when empty on the homepage template
1. Create page
2. Assign homepage template
3. Ensure it has an empty title / content
4. Publish
5. Notice a large empty section on the frontend

Something like 

```
.page-template-template-homepage .site-main > .type-page:empty { 
display: none; 
}
``` 

oughta do it.

### PR DESCRIPTION
**Pull request by tiagonoronha**
To test, edit the page with the "Homepage" template. If there's no content and title, the homepage section should not display in the frontend.

Fixes #673.

### CODE REVIEW

File: inc/storefront-template-functions.php [hunk 1]
```diff
@@ -845,8 +845,12 @@ function storefront_homepage_content() {
 		while ( have_posts() ) {
 			the_post();
 
-			get_template_part( 'content', 'homepage' );
+			$title   = get_the_title( get_the_ID() );
+			$content = get_the_content( get_the_ID() );
```
Comment (mikejolley) [inline inc/storefront-template-functions.php]: Incorrect usage of `get_the_content`. https://codex.wordpress.org/Function_Reference/get_the_content

File: inc/storefront-template-functions.php [hunk 2]
```diff
@@ -845,8 +845,12 @@ function storefront_homepage_content() {
 		while ( have_posts() ) {
 			the_post();
 
-			get_template_part( 'content', 'homepage' );
+			$title   = get_the_title( get_the_ID() );
```
Comment (mikejolley) [inline inc/storefront-template-functions.php]: No need to pass IDs in the loop afaik.

File: inc/storefront-template-functions.php [hunk 3]
```diff
@@ -845,8 +845,12 @@ function storefront_homepage_content() {
...
+			$content = get_the_content( get_the_ID() );
 
+			if ( '' !== $title || '' !== $content ) {
```
Comment (mikejolley) [inline inc/storefront-template-functions.php]: Maybe !empty would be cleaner. Nitpick.

**Comment by tiagonoronha** [comment]
I've been thinking about this one, I don't think I'm comfortable with this change.

For this to work you need an empty page title and empty content, then in WordPress you get the "no title" generic description for the page.

I think that's equally bad.

I'm going to close this PR.

**Pull request closed without merging**